src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -568,27 +576,43 @@ private async Task<IEnumerable<DiagnosticApiResponse>> ListDetectorsInternal(Run
             await this._sourceWatcherService.Watcher.WaitForFirstCompletion();
             var allDetectors = _invokerCache.GetEntityInvokerList<TResource>(context).ToList();
 
-            SearchResults searchResults = null;
             if (queryText != null && queryText.Length > 3)
             {
+                SearchResults searchResults = null;
                 var resourceParams = _internalApiHelper.GetResourceParams(context.OperationContext.Resource as IResourceFilter);
-                var res = await _searchService.SearchDetectors(context.OperationContext.RequestId, queryText, resourceParams);
-                string resultContent = await res.Content.ReadAsStringAsync();
-                searchResults = JsonConvert.DeserializeObject<SearchResults>(resultContent);
-                searchResults.Results = searchResults.Results.Where(x => x.Score > 0.3).ToArray();
-                allDetectors.ForEach(p =>
+                try
                 {
-                    var det = (searchResults != null) ? searchResults.Results.FirstOrDefault(x => x.Detector == p.EntryPointDefinitionAttribute.Id) : null;
-                    if (det != null)
+                    var res = await _searchService.SearchDetectors(context.OperationContext.RequestId, queryText, resourceParams);
+                    if (res != null && res.Content != null)
                     {
-                        p.EntryPointDefinitionAttribute.Score = det.Score;
+                        string resultContent = await res.Content.ReadAsStringAsync();
+                        searchResults = JsonConvert.DeserializeObject<SearchResults>(resultContent);
+                        searchResults.Results = searchResults.Results.Where(x => x.Score > 0.3).ToArray();
+                        allDetectors.ForEach(p =>
+                        {
+                            var det = (searchResults != null) ? searchResults.Results.FirstOrDefault(x => x.Detector == p.EntryPointDefinitionAttribute.Id) : null;
```

Comment:
Can we keep the variable names more easy to understand?

Reply:
Sure. Thanks!